dd-sdk-android/src/main/kotlin/com/datadog/android/rum/webview/RumEventConsumer.kt

```diff
@@ -6,13 +6,138 @@
 
 package com.datadog.android.rum.webview
 
+import com.datadog.android.core.internal.persistence.DataWriter
+import com.datadog.android.core.internal.time.TimeProvider
 import com.datadog.android.core.internal.utils.sdkLogger
+import com.datadog.android.rum.GlobalRum
+import com.datadog.android.rum.internal.domain.RumContext
+import com.datadog.android.rum.internal.domain.scope.RumApplicationScope
+import com.datadog.android.rum.internal.domain.scope.RumSessionScope
+import com.datadog.android.rum.internal.monitor.DatadogRumMonitor
+import com.datadog.android.rum.model.ActionEvent
+import com.datadog.android.rum.model.ErrorEvent
+import com.datadog.android.rum.model.LongTaskEvent
+import com.datadog.android.rum.model.ResourceEvent
+import com.datadog.android.rum.model.ViewEvent
 import com.google.gson.JsonObject
+import com.google.gson.JsonParseException
+import java.util.Locale.US
+import kotlin.collections.LinkedHashMap
 
-internal class RumEventConsumer {
+internal class RumEventConsumer(
+    private val dataWriter: DataWriter<Any>,
+    private val timeProvider: TimeProvider,
+    private val webRumEventMapper: WebRumEventMapper = WebRumEventMapper()
+) {
+
+    internal val offsets: LinkedHashMap<String, Long> = LinkedHashMap()
 
     fun consume(event: JsonObject, eventType: String) {
-        // just to make the code compile without warning. This will be removed in the next PR.
-        sdkLogger.i(event.toString() + eventType)
+        val rumContext = getRumContext()
+        if (rumContext != null &&
```

Comment:
should we emit any warning (or info) message otherwise?

Reply:
not sure honestly here, maybe a dev log ?

Reply:
This is for internal monitoring mostly, because I guess we would like to avoid such situations? And if it happens, we should have enough info to be able to find the root cause (if we consider it is an issue).

Reply:
Agreed, we should have: 
- A `devlog` to warn the customer that some WebView events will be lost
- Possibly an internal monitoring log if this issue can happen and mean something's wrong internally

Reply:
On a general basis, if we don't know something about code we use, it means we need to learn it :)